src/modules/scraper/adapter/messaging/TrackFillEventConsumer.ts

```diff
@@ -0,0 +1,94 @@
+import { OnQueueFailed, Process, Processor } from "@nestjs/bull";
+import { Logger } from "@nestjs/common";
+import { Job } from "bull";
+import { InjectRepository } from "@nestjs/typeorm";
+import { Repository } from "typeorm";
+import { utils } from "ethers";
+import { TrackFillEventQueueMessage, ScraperQueue } from ".";
+import { Deposit } from "../../model/deposit.entity";
+import { TrackService } from "../amplitude/track-service";
+import { makeWeiPctValuesFormatter } from "../amplitude/utils";
+import { ChainIdToName } from "../../../web3/model/ChainId";
+import BigNumber from "bignumber.js";
+
+@Processor(ScraperQueue.TrackFillEvent)
+export class TrackFillEventConsumer {
+  private logger = new Logger(TrackFillEventConsumer.name);
+
+  constructor(
+    private trackService: TrackService,
+    @InjectRepository(Deposit) private depositRepository: Repository<Deposit>,
+  ) {}
+
+  @Process()
+  private async process(job: Job<TrackFillEventQueueMessage>) {
+    const { depositId } = job.data;
+    const deposit = await this.depositRepository.findOne({ where: { id: depositId }, relations: ["token", "price"] });
+
+    if (!deposit) return;
+    if (deposit.status !== "filled") return;
+
+    if (!deposit.token || !deposit.price) {
+      throw new Error("Can not track fill event without token or price");
+    }
+
+    const amountFormatted = utils.formatUnits(deposit.amount, deposit.token.decimals);
+    const weiPctValuesFormatter = makeWeiPctValuesFormatter(amountFormatted, deposit.price.usd);
+    const formattedLpFeeValues = weiPctValuesFormatter(deposit.realizedLpFeePct);
+    const formattedRelayFeeValues = weiPctValuesFormatter(deposit.depositRelayerFeePct);
+    const formattedBridgeFeeValues = weiPctValuesFormatter(deposit.bridgeFeePct);
+
+    this.trackService.trackDepositFilledEvent(deposit.depositorAddr, {
+      capitalFeePct: "-",
+      capitalFeeTotal: "-",
+      capitalFeeTotalUsd: "-",
+      fromAmount: amountFormatted,
+      fromAmountUsd: new BigNumber(amountFormatted).multipliedBy(deposit.price.usd).toFixed(2),
+      fromChainId: String(deposit.sourceChainId),
+      fromChainName: ChainIdToName[deposit.sourceChainId],
+      fromTokenAddress: deposit.tokenAddr,
+      isAmountTooLow: false,
+      lpFeePct: formattedLpFeeValues.pct,
+      lpFeeTotal: formattedLpFeeValues.total,
+      lpFeeTotalUsd: formattedLpFeeValues.totalUsd,
+      NetworkFeeNative: "-",
+      NetworkFeeNativeToken: "-",
+      NetworkFeeUsd: "-",
```

Comment:
Not sure which transaction to take here in the case of multiple fill transactions 🤔 Also, do I need to call `providerDestinationChain.getTransaction(txHash)` to get these values?

Reply:
@nicholaspai

Reply:
shouldn't you track every fill transaction  since there can be many? And yes I think you can call [getTransaction](https://docs.ethers.org/v5/api/providers/provider/#Provider-getTransaction)

Reply:
Yea that's what I am not sure about. I understood that this event should track the event when a deposit gets fully filled. I will clarify with Ryan on this

Reply:
It got clarified and as you said all fill events should be tracked with the respective fill values.